cache/in-memory/src/event/mod.rs

```diff
@@ -29,9 +29,12 @@ impl InMemoryCache {
 
     fn cache_user(&self, user: Cow<'_, User>, guild_id: Option<GuildId>) {
         match self.users.get_mut(&user.id) {
-            Some(mut u) if u.0 == *user => {
+            Some(u) if *u.value() == *user => {
```

Comment:
why do we Deref both of these to compare them, can't we just compare references?

Reply:
`user` is a `Cow` and needs to be dereferenced, making the comparison `&User == `User`. `user` would need to be again referenced, resulting in `u.value() == &*user`. However, we can instead remove the symbol usage by using `user.as_ref()`. Done.